src/main/java/Duke.java

```diff
@@ -1,10 +1,212 @@
+import java.util.ArrayList;
+import java.util.Scanner;
+
+class Global {
+    static String PATTERNLINE = "____________________________________________________________________________";
+}
+
+enum Action {
+    Done,
+    List,
+    AddTodo,
+    AddDeadlines,
+    AddEvents,
+    Invalid,
+}
+
 public class Duke {
-    public static void main(String[] args) {
+
+    public static void main(String[] args) throws DukeException {
+
+        ArrayList<Task> tasks = new ArrayList<>();
+
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+
+        greetNote("welcome");
+
+        Scanner in = new Scanner(System.in);
+
+        while(in.hasNextLine()){
+            String input = in.nextLine();
+
+            if(!(input.toLowerCase()).equals("bye")) {
+                processInput(input, tasks);
+            } else {
+                greetNote("bye");
+                return;
+            }
+        }
+    }
+
+    private static void greetNote(String hi_bye) {
+
+        String welcome = Global.PATTERNLINE
+                        + "\n     Hello! I'm Duke\n"
+                        + "     What can I do for you?\n"
+                        + Global.PATTERNLINE + "\n";
+
+        String farewell = Global.PATTERNLINE
+                        + "\n   Bye. Hope to see you again soon!\n"
+                        + Global.PATTERNLINE;
+
+        if(hi_bye.equals("welcome")) {
+            System.out.println(welcome);
+        } else {
+            System.out.println(farewell);
+        }
+    }
+    private static void processInput(String input, ArrayList<Task> tasks) throws DukeException {
+
+        String command = input.toLowerCase();
+        Action action;// = Action.Invalid;
+        int count = tasks.size();
+
+        try {
+            action = validateCommand(command);
+        } catch (Exception g) {
+            System.out.println(g + "\n" + Global.PATTERNLINE);
+            return;
+        }
+
+        try {
+            switch(action) {
+            case List:
```

Comment:
I like this part as you abstract all actions to different methods. It's more readable and clear